trace-obfuscation/benches/credit_cards_bench.rs

```diff
@@ -44,4 +44,3 @@ fn is_card_number_no_luhn_bench(c: &mut Criterion) {
 }
 
 criterion_group!(benches, is_card_number_bench, is_card_number_no_luhn_bench);
-criterion_main!(benches);
```

Comment:
I'm a bit confused to how these things fit together. In `main` it's possible to run `cargo bench --bench credit_cards_bench` and it will run the these benchmarks only, but in this branch, they are skipped, and nothing is run.

Is the `benches_main` thing really necessary, and shouldn't the different bench files be defined in the `Cargo.toml` file?

Reply:
Hmm true, I added this because I found it annoying to edit the Cargo.toml file and was under the impression that it ran the linking part for each individual benchmark binary, which raised compile times when running all of them at once.

I was inspired by how the criterion-rs repo organizes their own benchmarks. https://github.com/bheisler/criterion.rs/tree/master/benches

You can still use `cargo bench -p datadog-trace-obfuscation card_number` to run the card number benches.

Reply:
Having a single bench binary also makes editing the Dockerfile for testing somewhat simpler since you don't have to add each of them individually https://github.com/DataDog/libdatadog/blob/main/tools/docker/Dockerfile.build#L106

Reply:
I don't see any changes to the docker file in this PR, so how would it change?

Reply:
I added the changes (the alpine test were failing without them)

Reply:
@paullegranddc I couldn't let got of the weird way that `cargo bench --bench` was displaying benches that doesn't exist (leading to confusion), and the solution is to move the benchmarks into a sub module in the same way as is done in the `criterion` repo https://github.com/bheisler/criterion.rs/tree/master/benches/benchmarks

Could we also rename the `benches_main` to `trace_obfuscation` to make it more clear where the `bench` lives? Then it would be possible to do things like `cargo bench --bench trace_obfuscation -- --list` to see all benchmarks available there.

Reply:
done